addons/hr_timesheet/controllers/portal.py

```diff
@@ -128,14 +128,15 @@ def portal_my_timesheets(self, page=1, sortby=None, filterby=None, search=None,
 
         def get_timesheets():
             field = None if groupby == 'none' else groupby
-            orderby = '%s, %s' % (field, sortby) if field else sortby
+            orderby = f'{field}, {sortby}' if field and field != sortby.split(' ')[0] else sortby
             timesheets = Timesheet_sudo.search(domain, order=orderby, limit=_items_per_page, offset=pager['offset'])
             if field:
                 if groupby == 'date':
                     raw_timesheets_group = Timesheet_sudo._read_group(
                         domain, ['date:day'], ['unit_amount:sum', 'id:recordset']
```

Comment:
I agree with Yannick, I think you just have to do this:
```suggestion
                        domain, ['date:day'], ['unit_amount:sum', 'id:recordset'], order='date:day desc',
```

Reply:
Thank you for the review, I took your suggestion and it did the job!